src/compiler/checker.ts

```diff
@@ -33961,16 +33961,10 @@ namespace ts {
 
             checkGrammarExportDeclaration(node);
             if (!node.moduleSpecifier || checkExternalImportOrExportDeclaration(node)) {
-                if (node.exportClause) {

```

Comment:
which part checks for the helper?

Reply:
Line 33985. Previously, `export * as ns` took the true side of the if statement on line 33964. Much of what what needed to happen (resolving the module specifier target, checking for helpers) was already happening in the false/else side, so the change here is primarily just routing `export * as ns` through the same code path as `export *`.